### PR TITLE
instrument label slicing

### PR DESCRIPTION
**Pull request by asher-pembroke**
# Description

Addresses # 262

`testInst[:,:]` was broken, so I added label slicing for instruments, which was causing tests to fail.

Modified the instrument test for pandas to slice values instead of index. 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update (possibly)

# How Has This Been Tested?
`pysat/tests/test_instrument.py` passes

**Test Configuration**:
* Mac 10.13.6
* miniconda python==3.7.3

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``master``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/_instrument.py [hunk 1]
```diff
@@ -443,7 +443,12 @@ def __getitem_xarray__(self, key):
                 try:
                     return self.data.isel(time=key[0])[key[1]]
                 except:
-                    return self.data.sel(time=key[0])[key[1]]
+                    try:
+                        return self.data.sel(time=key[0])[key[1]]
+                    except TypeError: # construct dataset from names
+                        variables = list(self.data.variables.keys())[key[1]]
+                        dataset_dict = {variable: self.data[variable] for variable in variables if variable not in self.data.coords}
+                        return xr.Dataset(dataset_dict, coords = self.data.coords, attrs = self.data.attrs)
```
Comment (rstoneback) [inline pysat/_instrument.py]: Can't we return the xarray dataset that already exists in data rather than recreate it?

**Comment by rstoneback** [comment]
Thanks for the code @asher-pembroke and the review @aburrell . We need a unit test for getting data using testInst[:, :] to go along with the code update. The basics for working with testInst[] are tested in tests/test_instrument.py.

File: pysat/_instrument.py (hunk 1 above)
Comment (asher-pembroke) [reply pysat/_instrument.py]: Yes, I think I have it right now. Didn't know you could do `dataset[list]`

**Comment by rstoneback** [comment]
I've confirmed test issues are related to the recently fixed F10.7 data change.

**Comment by rstoneback** [comment]
Merging into my pull.

**Pull request merged**